go-controller/pkg/cni/helper_linux.go

```diff
@@ -384,6 +374,79 @@ func (pr *PodRequest) ConfigureInterface(podLister corev1listers.PodLister, kcli
 	return []*current.Interface{hostIface, contIface}, nil
 }
 
+func (pr *PodRequest) UnconfigureInterface(ifInfo *PodInterfaceInfo) error {
+	podDesc := fmt.Sprintf("for pod %s/%s", pr.PodNamespace, pr.PodName)
+	klog.V(5).Infof("Tear down interface (%+v) %s", *pr, podDesc)
+	if pr.CNIConf.DeviceID == "" {
+		if ifInfo.IsDPUHostMode {
+			klog.Warningf("Unexpected configuration %s, Device ID must be present for pod request on smart-nic host",
+				podDesc)
+			return nil
+		}
+		// TODO(gmoodalbail): add similar check for the CX5 ASAP2 VF case
+	}
+
+	// For SRIOV case, we'd need to move the VF from container namespace back to the host namespace
+	if pr.CNIConf.DeviceID != "" {
```

Comment:
The two conditial checks on pr.CNIConf.DeviceID can be moved to single if..else..

Reply:
@zshi-redhat i have kept it separately because we need to also handle CX5 case. 

1st we check if DeviceID is set, then if we are running in DPU-Host mode, then we do things differently as compared to CX5 case